Services/LDAP/classes/class.ilLDAPAttributeToUser.php

```diff
@@ -290,21 +290,23 @@ private function usersToXML()
                 switch ($field) {
                     case 'gender':
                         switch (strtolower($value)) {
-                            case 'n':
-                            case 'neutral':
-                                $this->writer->xmlElement('Gender', array(), 'n');
-                                break;
-
                             case 'm':
                             case 'male':
                                 $this->writer->xmlElement('Gender', array(), 'm');
                                 break;
 
                             case 'f':
                             case 'female':
-                            default:
                                 $this->writer->xmlElement('Gender', array(), 'f');
                                 break;
+
+                            case 'n':
+                            case 'neutral':
+                            case 'd':
+                            case 'diverse':
```

Comment:
IMO we should not list `d` an `diverse` explicitly, just because there are other locations where we have to introduce these cases as well. IMO we should just use the `default` to handle other values.